### PR TITLE
QF-1203 Resolve security issue

### PR DESCRIPTION
**Pull request by SarahDuncan**
Edit code to be more robust against malicious inputs when running a stored proc against the database.

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Application.Api/Controllers/Staff/StaffReportsController.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ public async Task<IActionResult> GetReport(Guid reportId)
             }
             catch (SqlException sqlEx)
             {
-                _logger.LogInformation($"Could not generate report due to : {sqlEx.Message}");
+                _logger.LogError($"Could not generate data from report Id [{reportId}] due to : {sqlEx.Message}");
                 return NoContent();
```
Comment (SawNawDfE) [inline src/SFA.DAS.AssessorService.Application.Api/Controllers/Staff/StaffReportsController.cs:119]: Here we're logging an error, but returning a success status. That doesn't seem consistent
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

Comment (SarahDuncan) [reply src/SFA.DAS.AssessorService.Application.Api/Controllers/Staff/StaffReportsController.cs:119]: @ChrisJWoodcock  please see above

File: src/SFA.DAS.AssessorService.Data/Staff/StaffReportRepository.cs [hunk 2]
```diff
@@ -45,7 +44,14 @@ public Task<IEnumerable<StaffReport>> GetReportList()
 
         public async Task<IEnumerable<IDictionary<string, object>>> GetDataFromStoredProcedure(string storedProcedure)
         {
-            return (await _connection.QueryAsync(storedProcedure, commandType: CommandType.StoredProcedure)).OfType<IDictionary<string, object>>().ToList();
+            var reportId = _assessorDbContext.StaffReports.FirstOrDefault(rep => rep.StoredProcedure == storedProcedure.Trim());
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.AssessorService.Data/Staff/StaffReportRepository.cs:47]: Should rename to report

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AssessorService.Data/Staff/StaffReportRepository.cs (hunk 2 above)
Comment (SarahDuncan) [reply src/SFA.DAS.AssessorService.Data/Staff/StaffReportRepository.cs]: sorted

**Review by ChrisJWoodcock — APPROVED** [review]

**Pull request merged**